src/js/review/SubmitController.jsx

```diff
@@ -48,21 +48,27 @@ class SubmitController extends React.Component {
       form,
       formConfig,
       pagesByChapter,
-      privacyAgreementAccepted,
       trackingPrefix
     } = this.props;
 
-    const {
-      isValid,
-      errors
-    } = isValidForm(form, pagesByChapter);
+    let isValid;
+    let errors;
+
+    // If a pre-submit agreement was specified, it has to be accepted first
+    const preSubmitField = formConfig.preSubmitInfo &&
+        formConfig.preSubmitInfo.required && (formConfig.preSubmitInfo.field || 'AGREED');
+    if (preSubmitField && !form.data[preSubmitField]) {
```

Comment:
If we left that check in `isValidForm` we'd need to pass in the information about the preSubmit checkbox and its value just so it could check them. If we weren't treating this checkbox specially, for example if it were just a plain required checkbox with an error message, I think it would be natural to let `isValidForm` deal with it. However, since this is a special component that is actually part of SubmitController it seems unnatural to pass this information to a utility function to check it. It didn't stick out as much before since the name of the field and the value (`true`) was hard-wired, but it still was a special case.